erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -393,6 +393,10 @@ def set_unallocated_amount(self):
 			if self.payment_type == "Receive" \
 				and self.base_total_allocated_amount < self.base_received_amount + total_deductions \
 				and self.total_allocated_amount < self.paid_amount + (total_deductions / self.source_exchange_rate):
+					# Nonetype throws type error while performing mathematical manipulation thus assigned a value zero 
+					if self.base_total_discounted_amount is None:
+						self.base_total_discounted_amount = 0
+					
```

Comment:
This is over intended by 1 tab, fix the indentation for rest of this function as well